apache/superset
# fix: Reset sorting bar issue in Barchart

### SUMMARY

When enabling "sort bars" and disabling again, the order isn't reset to the original order.

It was because when we sort data from props, it made updates in redux store as well. 

Solution: deep copy

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

BEFORE:
<a href="https://www.loom.com/share/818b251b798946fa8fad094b95e34073">
    <p>[DEV] Genders by State - 25 March 2022 - Watch Video</p>
    <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/818b251b798946fa8fad094b95e34073-with-play.gif">
  </a>

AFTER:
<a href="https://www.loom.com/share/75005f3a14b24d63997bf2a13c8c8a6c">
    <p>[DEV] Genders by State - 25 March 2022 - Watch Video</p>
    <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/75005f3a14b24d63997bf2a13c8c8a6c-with-play.gif">
  </a>

### TESTING INSTRUCTIONS

1. Goto charts page
2. Select Genders by State chart
3. Check sort bars under customize tab.
4. Uncheck sort bars.
5. See how chart is rendered.

### ADDITIONAL INFORMATION


- [x] Has associated issue: https://github.com/apache/superset/issues/12650
